src/common/sockman.cpp

```diff
@@ -297,3 +378,151 @@ void SockMan::ThreadI2PAccept()
         err_wait = err_wait_begin;
     }
 }
+
+void SockMan::ThreadSocketHandler()
+{
+    AssertLockNotHeld(m_connected_mutex);
+
+    while (!interruptNet) {
+        EventIOLoopCompletedForAllPeers();
+
+        // Check for the readiness of the already connected sockets and the
+        // listening sockets in one call ("readiness" as in poll(2) or
+        // select(2)). If none are ready, wait for a short while and return
+        // empty sets.
+        auto io_readiness{GenerateWaitSockets()};
+        if (io_readiness.events_per_sock.empty() ||
+            !io_readiness.events_per_sock.begin()->first->WaitMany(SELECT_TIMEOUT,
```

Comment:
c133a634d10e172652ebbf05ca114037c8e551b9

You might want to add another comment here - when I was trying to handle socks myself I found this a bit weird and wrote:

```
    // WaitMany() mine as well be a static function, the context
    // of the first Sock in the vector is not relevant.
```

Reply:
Done